optimize/backend/src/main/java/io/camunda/optimize/tomcat/SaasRequestAdjustmentFilter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.tomcat;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.RequestDispatcher;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+public class SaasRequestAdjustmentFilter implements Filter {
+
+  private final String clusterId;
+
+  public SaasRequestAdjustmentFilter(final String clusterId) {
+    String cid = clusterId;
+    if (cid == null) {
+      cid = "";
+    }
+    this.clusterId = cid;
+  }
+
+  @Override
+  public void doFilter(
+      final ServletRequest request, final ServletResponse response, final FilterChain chain)
+      throws IOException, ServletException {
+    boolean forward = false;
+    final HttpServletRequest httpRequest = (HttpServletRequest) request;
+    String requestURI = httpRequest.getRequestURI();
+
+    /* Strip cluster ID */
+    if (!clusterId.isEmpty() && requestURI.startsWith("/" + clusterId)) {
+      requestURI = requestURI.substring(("/" + clusterId).length());
+      forward = true;
+    }
+
+    /* transform /external/api -> /api/external */
```

Comment:
the reason why I'm applying 2 transformation in the same filter (instead of having a common externalApiRewriter + the cluster ID filter) is because after the first transformation+forward, the forwarded request doesn't go through the filter chain again